### PR TITLE
NOISSUE - Provide API interaction examples

### PR DESCRIPTION
**Pull request by anovakovic01**
Update documentation with example requests and responses in accordance
with new API changes. Add detailed flow explanation.

Signed-off-by: Aleksandar Novakovic <anovakovic01@gmail.com>

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -17,11 +17,25 @@ docker-compose up -f docker/docker-compose.yml -d
 
 ### Account creation
 
-TBD
+Use the Mainflux manager HTTP/REST API to create user account:
+
+```
+curl -s -S -i -X POST -H "Content-Type: application/json; charset=utf-8" localhost:8180/users -d '{"email":"john.doe@email.com", "password":"123"}'
+```
```
Comment (drasko) [inline docs/getting-started.md]: I would also add `HTTPS` variant served by NginX (i.e. port `443` or `80`->`443` redirection)

Comment (drasko) [reply docs/getting-started.md]: It does not have to be for every example, but at least one - to show people how to use HTTPS + cert

**Review by drasko — APPROVED** [review]
LGTM

File: docs/getting-started.md [hunk 2]
```diff
@@ -9,19 +9,37 @@ Before proceeding, install the following prerequisites:
...
-cd docker/
 docker-compose up -f docker/docker-compose.yml -d
 ```
 
 ## User management
 
 ### Account creation
 
-TBD
+Use the Mainflux manager HTTP/REST API to create user account:
```
Comment (mijicd) [inline docs/getting-started.md]: `Use the Mainflux API to create user account:`. Drop REST/HTTP wherever not necessary.

File: docs/getting-started.md [hunk 3]
```diff
@@ -30,35 +48,196 @@ an authorization key.
...
+Devices are provisioned by executing HTTP request `POST /clients`, with a 
+`"type":"device"` specified in JSON payload. Note that you will also need 
+`user_auth_token` in order to provision clients (both devices and application) 
+that belong to this particular user.
+
+```
+curl -s -S -i --cacert docker/ssl/certs/mainflux-server.crt --insecure -X POST -H "Content-Type: application/json; charset=utf-8" -H "Authorization: <user_auth_token>" https://localhost/clients -d '{"type":"device", "name":"weio"}'
+```
+
+Response will contain `Location` header which value represents path to newly
```
Comment (mijicd) [inline docs/getting-started.md]: `... header whose value ...`

File: docs/getting-started.md [hunk 4]
```diff
@@ -30,35 +48,196 @@ an authorization key.
...
+Location: /clients/81380742-7116-4f6f-9800-14fe464f6773
+Date: Tue, 10 Apr 2018 10:02:59 GMT
+Content-Length: 0
+```
 
 ### Provisioning applications
 
-TBD
+Applications are provisioned by executing HTTP request `POST /clients`, with 
+`"type":"app"` specified in JSON payload. Note that applications are also Mainflux
```
Comment (mijicd) [inline docs/getting-started.md]: No need for `note that` section.

File: docs/getting-started.md [hunk 5]
```diff
@@ -30,35 +48,196 @@ an authorization key.
...
-TBD
+Applications are provisioned by executing HTTP request `POST /clients`, with 
+`"type":"app"` specified in JSON payload. Note that applications are also Mainflux 
+clients, as are devices.
+
+```
+curl -s -S -i --cacert docker/ssl/certs/mainflux-server.crt --insecure -X POST -H "Content-Type: application/json; charset=utf-8" -H "Authorization: <user_auth_token>" https://localhost/clients -d '{"type":"app", "name":"myapp"}'
+```
+
+Response will contain `Location` header which value represents path to newly
```
Comment (mijicd) [inline docs/getting-started.md]: `... header whose value ...`

**Review by mijicd — CHANGES_REQUESTED** [review]

**Review by mijicd — APPROVED** [review]

**Pull request merged**